### ISSUE TITLE
Custom duration isn't loaded from the URL

### ISSUE THREAD

**Issue by elie222** [calcom/cal.com#6433]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

My link is:
https://cal.com/abc/my-meeting?duration=60
Ideally when the page loads it would select 60 minutes but it still selects my default of 30 minutes.

### Steps to Reproduce

1. Create an event with "Allow booker to select duration". Add two durations
2. Go to event link and select the non default link. The url will now update
3. Refresh the page - the selected slot is still the default and not the one from the URL.

**Comment by PeerRich** [comment]
I don't think we have a parameter for duration yet. let me investigate. tagging @leog who built multiple durations

**Comment by elie222** [comment]
You do have the parameter in the URL. But the page doesn't do anything with that param.

**Comment by leog** [comment]
Weird behavior on the UI, the actual duration is set correctly, check your slots. Will take a look. Thanks for reporting it @elie222.

**Comment by jahirfiquitiva** [comment]
What's happening on my end is that the time option technically is selected, but not visually.
I have to select another time option and then the other one so that is actually appears selected.

https://user-images.githubusercontent.com/10360816/224430630-7dfd543b-4146-452c-9ad9-284fd6fef119.mp4

**Comment by PeerRich** [comment]
@leog can you reinvestigate this?

**Comment by leog** [comment]
This should be fixed by #10513
